From f8acbbb0ac154fa812e6297e86fad58381d4a163 Mon Sep 17 00:00:00 2001
From: Tomasz Zawadzki <tomasz.zawadzki@intel.com>
Date: Fri, 12 Feb 2021 03:45:16 -0500
Subject: [PATCH 06/15] lib/env_dpdk: add rte_net dependency

Vhost is enabled by default, so rte_net was always included.
When disabled, rte_power failed as it depends on rte_ethdev and rte_net.

rte_vhost was only possible to enable on Linux, so there
is no conflict with adding it next to rte_power under this condition.

Signed-off-by: Tomasz Zawadzki <tomasz.zawadzki@intel.com>
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/spdk/+/6398 (master)

(cherry picked from commit 232f34d754e08b9673fd3585786ca8cd55653b85)
Change-Id: I2e183004d6457e404471740a0540dcb08aa738d8
Signed-off-by: Tomasz Zawadzki <tomasz.zawadzki@intel.com>
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/spdk/+/6430
Tested-by: SPDK CI Jenkins <sys_sgci@intel.com>
Reviewed-by: Jim Harris <james.r.harris@intel.com>
Reviewed-by: Ben Walker <benjamin.walker@intel.com>
---
 lib/env_dpdk/env.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/env_dpdk/env.mk b/lib/env_dpdk/env.mk
index f44f515..292dd91 100644
--- a/lib/env_dpdk/env.mk
+++ b/lib/env_dpdk/env.mk
@@ -51,7 +51,7 @@ DPDK_INC := -I$(DPDK_INC_DIR)
 DPDK_LIB_LIST = rte_eal rte_mempool rte_ring rte_mbuf rte_pci rte_bus_pci rte_mempool_ring
 
 ifeq ($(OS),Linux)
-DPDK_LIB_LIST += rte_power rte_ethdev
+DPDK_LIB_LIST += rte_power rte_ethdev rte_net
 endif
 
 # DPDK 20.05 eal dependency
@@ -101,7 +101,7 @@ endif
 LINK_HASH=n
 
 ifeq ($(CONFIG_VHOST),y)
-DPDK_LIB_LIST += rte_vhost rte_net
+DPDK_LIB_LIST += rte_vhost
 LINK_HASH=y
 ifneq ($(DPDK_FRAMEWORK),y)
 DPDK_LIB_LIST += rte_cryptodev
-- 
1.8.3.1

